plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordSetProvider.java

```diff
@@ -54,7 +55,7 @@ public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorS
         // If no columns are recorded, it means that applyProjection never got called (e.g., in the case all columns are being used) and all
         // table columns should be returned. TODO: this is something that should be addressed once the getRecordSet API is revamped
         jdbcTable.getColumns()
-                .ifPresent(tableColumns -> verify(columns.equals(tableColumns)));
+                .ifPresent(tableColumns -> verify(ImmutableSet.copyOf(tableColumns).containsAll(columns)));
```

Comment:
- update the comment above (the first line: "(and order)")
- i don't know much about "TODO: this is something that should be addressed once the getRecordSet API" but @martint may want to update it as a followup

Reply:
I think the "order" in comment is still valid. It is about requirements regarding output of JdbcRecordSet. Not about the assertion itself.

Reply:
You're right. Maybe we should update the first sentence (or add second) recording that order doesn't matter

Reply:
Reworded a bit to not suggest that exact match is required.